elf/segmentation/watershed.py

```diff
@@ -25,6 +25,12 @@ def watershed(input_, seeds, size_filter=0, exclude=None):
         np.ndarray - watershed segmentation
         int - max id of watershed segmentation
     """
+
+    if input_.dtype != 'uint8' or seeds.dtype != 'uint32':
```

Comment:
Vigra supports `float32` inputs as well.
Also, check out https://stackoverflow.com/questions/26921836/correct-way-to-test-for-numpy-dtype for proper dtype checking.

Reply:
Thanks, that's a useful answer!